### ISSUE TITLE
[MaterialDatePicker] OnSelectionChangedListener and DateSelector are not accessible.

### ISSUE THREAD

**Issue by kevintorch** [material-components/material-components-android#1727]
**Description:** Trying to implement custom DateSelector. but it shows error as

**_"com.google.android.material.datepicker.OnSelectionChangedListener' is not public in 'com.google.android.material.datepicker'. Cannot be accessed from outside package"_**

and

**_"DateSelector can only be accessed from within the same library group (groupId=com.google.android.material)"_**

**Expected behavior:** Basically I want the same behavior as SingleDateSelector or RangeDateSelector

**Material Library version:** 1.3.0-alpha02

**Comment by matpag** [comment]
If you need those specific functionalities you can use the builder:
```
//Range selector
MaterialDatePicker.Builder<Pair<Long, Long>> builder = MaterialDatePicker.Builder.dateRangePicker();
```
or
```
//Single selector
MaterialDatePicker.Builder<Long> builder = MaterialDatePicker.Builder.datePicker();
```

OT: Maybe `DateSelector` should be opened and everyone should be able to implement a custom one if needed.
Obviously the builder should have an additional `MaterialDatePicker.Builder.withPicker(customDateSelector);`

I don't know if implementing a new custom `DateSelector` it's an "easy" task...

**Comment by kevintorch** [comment]
> Maybe `DateSelector` should be opened and everyone should be able to implement a custom one if needed.
> Obviously the builder should have an additional `MaterialDatePicker.Builder.withPicker(customDateSelector);`
> 
> I don't know if implementing a new custom `DateSelector` it's an "easy" task...

Actually it has MaterialDatePicker.Builder.customSelector(DateSelector) method. but it's no use.

and second, it's not that difficult to implement a custom DateSelector. I've seen the classes

**Comment by matpag** [comment]
You are right, it's this, https://github.com/material-components/material-components-android/blob/22973ff10305397bfc72fed41684175fee8aeefb/lib/java/com/google/android/material/datepicker/MaterialDatePicker.java#L548
but it's Library Group restricted as everything else.

> and second, it's not that difficult to implement a custom DateSelector. I've seen the classes
Nice to know 👍

**Comment by AleksandarAleksiev** [comment]
I have the same issue.  I need to listen on selection changes and need to add custom logic for`DateSelector:isSelectionComplete` but 🤷‍♂️ no possible at the moment.

**Comment by lsuski** [comment]
E.g this could be used to implement optional range selection

**Comment by saoudiabderraouf** [comment]
Hello everyone, actually I have the same issue. I need to listen on selection changes and need to disable some dates when first date is selected but no possible at the moment. I hope they add a listener.

**Comment by sme-shyl001** [comment]
Still not fixed :(
